rpcserver.go

```diff
@@ -1684,6 +1701,31 @@ func (r *rpcServer) OpenChannel(in *lnrpc.OpenChannelRequest,
 			if err != nil {
 				return err
 			}
+
+		case in.FundingShim.GetPsbtShim() != nil:
+
+			// The funding intent must be present.
+			psbtShim := in.FundingShim.GetPsbtShim()
+			copy(req.pendingChanID[:], psbtShim.PendingChanId)
+			intent, err := r.server.cc.wallet.GetFundingIntent(
```

Comment:
So we require the user to already have registered the intent? Usually we require registration only when we're the _responder_ to a special funding flow. Instead, I would think this step would just set the channel funder to the PSBT _assembler_ itself.

Reply:
Ah, I did indeed ask myself why the registration of the shim was needed prior to opening a channel when we send it there again. I didn't think of the case were we _aren't_ the initiator and still need a way to inform the funding manager about the incoming channel. It makes much more sense now and I removed the requirement for the shim to be registered before calling `OpenChannel`.